src/Components/cards/card.jsx

```diff
@@ -2,14 +2,19 @@ import React from "react";
 import { useProduct } from "../../context/product-context"
 import { favourite_icon } from "../../Assets/index";
 import "./card.css";
+import { filterByCategory, RatingFilter, SortingFilter } from "../../reducers/filter";
 function ProductCard() {
-  const { items } = useProduct();
- 
+  const { items,state } = useProduct();
+  const {filter}=state
+  const{ratings,sortBy,categoryName}=filter
+  const SortedData=SortingFilter(items,sortBy)
+  const CategoryData=filterByCategory(SortedData,categoryName)
+ const RatingData=RatingFilter(CategoryData,ratings)
 
   return (
-    <div>
-      {items.map((item) => (
-        <div className="product-grid">
+    <div className="product-grid">
+      {RatingData.map((item) => (
+        <div className="">
```

Comment:
One small suggestion. Inside the map, the function pass key attribute otherwise given a warning message in the console. Please check it once.